src/BoundaryConditions/boundary_condition.jl

```diff
@@ -32,23 +32,24 @@ Adapt.adapt_structure(to, b::BoundaryCondition{C, A}) where {C<:BCType, A<:Abstr
 const BC   = BoundaryCondition
 const FBC  = BoundaryCondition{<:Flux}
 const PBC  = BoundaryCondition{<:Periodic}
-const NPBC = BoundaryCondition{<:NoPenetration}
+const NFBC = BoundaryCondition{<:NormalFlow}
 const VBC  = BoundaryCondition{<:Value}
 const GBC  = BoundaryCondition{<:Gradient}
-const NFBC = BoundaryCondition{Flux, Nothing}
+const ZFBC = BoundaryCondition{Flux, Nothing} # "zero" flux
 
 # More readable BC constructors for the public API.
-     PeriodicBoundaryCondition() = BoundaryCondition(Periodic,      nothing)
-NoPenetrationBoundaryCondition() = BoundaryCondition(NoPenetration, nothing)
-       NoFluxBoundaryCondition() = BoundaryCondition(Flux,          nothing)
+    PeriodicBoundaryCondition() = BoundaryCondition(Periodic,   nothing)
+      NoFluxBoundaryCondition() = BoundaryCondition(Flux,       nothing)
```

Comment:
Should we rename this to `ZeroFluxBoundaryCondition` to be consistent with `ZFBC`?

Reply:
I guess `ZFBC` is internal whereas this is API. Do we want to change the API?

Reply:
@ali-ramadhan thoughts?